coderd/inboxnotifications.go

```diff
@@ -0,0 +1,376 @@
+	if err != nil {
+		httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+			Message: "Invalid query parameter.",
+			Detail:  err.Error(),
+		})
+		return
+	}
+
+	startingBefore := dbtime.Now()
+	if startingBeforeParam != "" {
+		lastNotifID, err := uuid.Parse(startingBeforeParam)
+		if err != nil {
+			httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+				Message: "Invalid starting before.",
+			})
+			return
+		}
+		lastNotif, err := api.Database.GetInboxNotificationByID(ctx, lastNotifID)
+		if err != nil {
+			httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+				Message: "Failed to get notification by id.",
+			})
+			return
+		}
+		startingBefore = lastNotif.CreatedAt
+	}
+
+	notifs, err := api.Database.GetFilteredInboxNotificationsByUserID(ctx, database.GetFilteredInboxNotificationsByUserIDParams{
+		UserID:       apikey.UserID,
+		Templates:    templates,
+		Targets:      targets,
+		ReadStatus:   database.InboxNotificationReadStatus(readStatus),
+		CreatedAtOpt: startingBefore,
+	})
+	if err != nil {
+		api.Logger.Error(ctx, "failed to get filtered inbox notifications", slog.Error(err))
+		httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+			Message: "Failed to get filtered inbox notifications.",
+		})
+		return
+	}
+
+	unreadCount, err := api.Database.CountUnreadInboxNotificationsByUserID(ctx, apikey.UserID)
+	if err != nil {
+		api.Logger.Error(ctx, "failed to count unread inbox notifications", slog.Error(err))
+		httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+			Message: "Failed to count unread inbox notifications.",
+		})
+		return
+	}
+
+	httpapi.Write(ctx, rw, http.StatusOK, codersdk.ListInboxNotificationsResponse{
+		Notifications: func() []codersdk.InboxNotification {
+			notificationsList := make([]codersdk.InboxNotification, 0, len(notifs))
+			for _, notification := range notifs {
+				notificationsList = append(notificationsList, convertInboxNotificationResponse(ctx, api.Logger, notification))
+			}
+			return notificationsList
+		}(),
+		UnreadCount: int(unreadCount),
+	})
+}
+
+// updateInboxNotificationReadStatus changes the read status of a notification.
+// @Summary Update read status of a notification
+// @ID update-read-status-of-a-notification
+// @Security CoderSessionToken
+// @Produce json
+// @Tags Notifications
+// @Param id path string true "id of the notification"
+// @Success 201 {object} codersdk.Response
+// @Router /notifications/inbox/{id}/read-status [put]
+func (api *API) updateInboxNotificationReadStatus(rw http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	var (
+		apikey  = httpmw.APIKey(r)
+		notifID = chi.URLParam(r, "id")
+	)
+
+	var body codersdk.UpdateInboxNotificationReadStatusRequest
+	if !httpapi.Read(ctx, rw, r, &body) {
+		return
+	}
+
+	parsedNotifID, err := uuid.Parse(notifID)
```

Comment:
This can also be handled in the request parsing using `format:"uuid"`.

Reply:
+1